scripts/run-test.sh

```diff
@@ -3,7 +3,8 @@
 # Due to async/await being used in tests, we can't run node version < 8
 echo "Checking node version..."
 node --version | grep v8
-if [[ $? != 0 ]] ; then
+RETURN_CODE=$?
+if [ "${RETURN_CODE}" -ne "0" ]; then
```

Comment:
@jmelberg-okta: Since `.travis.yml` has node 6, if we exit with 1, the build will be marked as failure. This is just a workaround to not fail the build when node version is 6.

Reply:
With this approach its all-or-nothing though, right? For example:
  -  Run the suite for node 8 and output `success` or `fail`
  - For all other versions outside of 8, return `success`

This seems like we're placing the ignore logic incorrectly. Since we have a few tests that fail outside of node 8, we should *ignore* **only** those tests.

Thus,

```javascript
nodeCheck('throws an error if blah', function () {
  // Do assertions
});

function nodeCheck(test) {
  if (process.env.NODE_VERSION === '8') {
    return it(test);
  } else {
  // Node version isn't supported, ignore test
   console.warn('Ignoring test due to incompatible Node version');
   itx(test);
  }
}
```

Reply:
Following up with the above comment, it seems like we're using the `async`, `await` syntax nearly everywhere in our integration tests. So we do not really have a choice but to ignore 
it all.

I now see why you were attempting to ignore the entire suite. 😄 

We should probably file a JIRA to migrate these tests to support Node 4.x.